bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java

```diff
@@ -65,28 +67,22 @@
  *
  * @author Markus Eckhardt - Initial contribution
  */
+@NonNullByDefault
 public class KM200ThingHandler extends BaseThingHandler {
 
     private final Logger logger = LoggerFactory.getLogger(KM200ThingHandler.class);
-    private static URI configDescriptionUriChannel;
 
     public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.unmodifiableSet(Stream
             .of(THING_TYPE_DHW_CIRCUIT, THING_TYPE_HEATING_CIRCUIT, THING_TYPE_SOLAR_CIRCUIT, THING_TYPE_HEAT_SOURCE,
                     THING_TYPE_SYSTEM_APPLIANCE, THING_TYPE_SYSTEM_HOLIDAYMODES, THING_TYPE_SYSTEM_SENSOR,
                     THING_TYPE_GATEWAY, THING_TYPE_NOTIFICATION, THING_TYPE_SYSTEM, THING_TYPE_SYSTEMSTATES)
             .collect(Collectors.toSet()));
 
-    private KM200ChannelTypeProvider channelTypeProvider;
+    private @Nullable KM200ChannelTypeProvider channelTypeProvider;
 
-    public KM200ThingHandler(Thing thing, KM200ChannelTypeProvider channelTypeProvider) {
+    public KM200ThingHandler(Thing thing, @Nullable KM200ChannelTypeProvider channelTypeProvider) {
```

Comment:
Once you make the `KM200ChannelTypeProvider` non-null and final in the ThingHandlerFactory then you should be able to make the non-null again.

Reply:
Ok, changed.